tutorials/backend/hasura-v3-ts-connector/tutorial-site/content/setup-connector.md

```diff
@@ -0,0 +1,607 @@
+      fields
+    };
+  }
+
+  let collections: CollectionInfo[] = configuration.tables.map((table) => {
+    return {
+      arguments: {},
+      name: table.tableName,
+      deletable: false,
+      foreign_keys: {},
+      uniqueness_constraints: {},
+      type: table.tableName,
+    };
+  });
+}
+```
+
+Again, we define one collection per table in the configuration, and we use the object type with the same name that we
+just defined.
+
+Now we can put the schema response together:
+
+```typescript
+async function get_schema(configuration: RawConfiguration): Promise<SchemaResponse> {
+  let scalar_types: { [k: string]: ScalarType } = {
+    'any': {
+      aggregate_functions: {},
+      comparison_operators: {},
+      update_operators: {},
+    }
+  };
+
+  let object_types: { [k: string]: ObjectType } = {};
+
+  for (const table of configuration.tables) {
+    let fields: { [k: string]: ObjectField } = {};
+
+    for (const columnName in table.columns) {
+      fields[columnName] = {
+        type: {
+          type: 'named',
+          name: 'any'
+        }
+      };
+    }
+
+    object_types[table.tableName] = {
+      fields
+    };
+  }
+
+  let collections: CollectionInfo[] = configuration.tables.map((table) => {
+    return {
+      arguments: {},
+      name: table.tableName,
+      deletable: false,
+      foreign_keys: {},
+      uniqueness_constraints: {},
+      type: table.tableName,
+    };
+  });
+
+  return {
+    functions: [],
+    procedures: [],
+    collections,
+    object_types,
+    scalar_types,
+  };
+}
+```
+
+Notice that we don't define `functions` or `procedures`, but we'll cover those features later in the series.
+
+## Testing
+
+So we have one more function to define, which is the query function, but before we do, let's talk about tests. The [NDC
+specification repository](https://github.com/hasura/ndc-spec/) provides a 
+[test runner executable](https://github.com/hasura/ndc-spec/tree/main/ndc-test) called `ndc-test`, which can be used to 
+implement a test suite for a connector. 
+
+We can also use ndc-test to run some automatic tests and validate the work we've done so far. Let's compile and run 
+our connector, and then use the test runner with the running connector.
+
+Here I have a configuration.json file which I can use to run the connector against my sample database.
+
+First let's run the connector.
+
+```shell
+npm run build && node dist/index.js serve --configuration configuration.json
+```
+
+[//]: # (TODO - document the test runner better in the spec repo)
+Now, let's run the tests. (You will need to have the 
+[ndc test runner](https://github.com/hasura/graphql-engine-mono/pull/10626#pullrequestreview-1826948556) installed on 
+your machine.)
+
+```shell
+ndc-test test --endpoint http://localhost:8100
+```
+
+OR
+
+```shell
+cargo run --bin ndc-test -- test --endpoint http://localhost:8100
+````
+
+Of course, we expected it to fail, but we can already see that our schema response is good.
```

Comment:
Can we provide the expected output of the test runner for a user to compare what they get? When I run the test with the `index.ts` as is up to this point, I get `OK` across the board. My connector also returns the schema as you said, but I'm struggling to see where the `failure` is coming from.